### PR TITLE
Fix bias dtype

### PR DESCRIPTION
**Pull request by wenscarl**
## 📌 Description

co-authored by @ChristinaZ(major contribution)
After this change, for deepseek mode, now logits is always fp32, bias can be bfloat16 and fp32 now.

## 🚀 Pull Request Checklist

Thank you for contributing to FlashInfer! Before we review your pull request, please make sure the following items are complete.

### ✅ Pre-commit Checks

- [ ] I have installed `pre-commit` by running `pip install pre-commit` (or used your preferred method).
- [ ] I have installed the hooks with `pre-commit install`.
- [ ] I have run the hooks manually with `pre-commit run --all-files` and fixed any reported issues.

> If you are unsure about how to set up `pre-commit`, see [the pre-commit documentation](https://pre-commit.com/).

## 🧪 Tests

- [ ] Tests have been added or updated as needed.
- [ ] All tests are passing (`unittest`, etc.).

### CODE REVIEW

File: csrc/trtllm_fused_moe_kernel_launcher.cu [hunk 1]
```diff
@@ -109,6 +109,10 @@ Tensor trtllm_fp8_per_tensor_scale_moe_launcher(
   args.routing_logits = routing_logits->data;
   auto const routing_bias_dtype =
       routing_bias.has_value() ? routing_bias.value()->dtype : dl_bfloat16;
```
Comment (pavanimajety) [inline csrc/trtllm_fused_moe_kernel_launcher.cu:113]: When would `routing_bias` not have value? Can we combine the statements and ensure that we always follow the dtype specified as the `routing_bias` tensor input's type?

File: csrc/trtllm_fused_moe_runner.cu [hunk 2]
```diff
@@ -55,13 +55,18 @@ void Runner::run(void* routingLogits, void* routingBias, int32_t numTokens, int3
...
+                 int32_t* numNonExitingCtas, btg::Dtype dtypeElt, btg::Dtype dtypeBias,
+                 bool useRoutingScalesOnInput, bool useDeepSeekFp8,
+                 RoutingMethodType routingMethodType, cudaStream_t stream) {
   if (routingMethodType == RoutingMethodType::DeepSeekV3) {
     FLASHINFER_CHECK(topK <= 8, "For DeepSeek routing method, must have topK <= 8");
     FLASHINFER_CHECK(topkGroup <= 4, "For DeepSeek routing method, must have topkGroup <= 4");
     moe::dev::routing::routingDeepSeek::Data routingData;
-    routingData.mDtypeExpW = btg::Dtype::Bfloat16;
+    routingData.mDtypeExpW =
+        btg::Dtype::Bfloat16;            // for DeepSeek, the expW is currently always bfloat16
```
Comment (pavanimajety) [inline csrc/trtllm_fused_moe_runner.cu:66]: what is expW?

File: csrc/trtllm_fused_moe_kernel_launcher.cu [hunk 3]
```diff
@@ -184,8 +190,9 @@ Tensor trtllm_fp8_per_tensor_scale_moe_launcher(
       static_cast<int*>(num_tokens_per_expert->data),
       static_cast<int*>(cta_idx_xy_to_batch_idx->data),
       static_cast<int*>(cta_idx_xy_to_mn_limit->data),
-      static_cast<int*>(num_non_exiting_ctas->data), args.mDtypeElt, use_routing_scales_on_input,
-      false /* use_deep_seek_fp8 */, static_cast<RoutingMethodType>(routing_method_type), stream);
+      static_cast<int*>(num_non_exiting_ctas->data), args.mDtypeElt, btg_routing_bias_dtype,
+      use_routing_scales_on_input, false /* use_deep_seek_fp8 */,
```
Comment (pavanimajety) [inline csrc/trtllm_fused_moe_kernel_launcher.cu:194]: does use_deep_seek_fp8=False mean per-tensor FP8?

File: csrc/trtllm_fused_moe_kernel_launcher.cu [hunk 4]
```diff
@@ -432,20 +443,20 @@ void trtllm_fp8_block_scale_moe_launcher(
...
+      static_cast<int*>(expert_count_histogram->data),
+      static_cast<int*>(total_num_padded_tokens->data),
+      static_cast<int*>(expanded_idx_to_permuted_idx->data),
+      nullptr /*static_cast<int*>(permuted_idx_to_expanded_idx->data)*/,
+      static_cast<int*>(permuted_idx_to_token_idx->data), expert_weights->data,
+      static_cast<int*>(num_tokens_per_expert->data),
+      static_cast<int*>(cta_idx_xy_to_batch_idx->data),
+      static_cast<int*>(cta_idx_xy_to_mn_limit->data),
+      static_cast<int*>(num_non_exiting_ctas->data), args.mDtypeElt, btg_routing_bias_dtype, false,
+      true, static_cast<RoutingMethodType>(routing_method_type), stream);
```
Comment (pavanimajety) [inline csrc/trtllm_fused_moe_kernel_launcher.cu]: Nit: Can we mention use_routing_scales_on_input and use_deep_seek_fp8 here as well as comments?

**Comment by pavanimajety** [comment]
@ChristinaZ  Thanks for making changes across all MOE runners!

File: csrc/trtllm_fused_moe_runner.cu (hunk 2 above)
Comment (ChristinaZ) [reply csrc/trtllm_fused_moe_runner.cu:66]: For the routing part, it's the output data type of the routing kernel.

File: csrc/trtllm_fused_moe_kernel_launcher.cu (hunk 1 above)
Comment (ChristinaZ) [reply csrc/trtllm_fused_moe_kernel_launcher.cu:113]: When the routing method is not deepseek (for example, renormalize routing method), the `routing_bias` does not have value.

File: csrc/trtllm_fused_moe_kernel_launcher.cu (hunk 3 above)
Comment (ChristinaZ) [reply csrc/trtllm_fused_moe_kernel_launcher.cu:194]: I think  `use_deep_seek_fp8=True ` means the blockscale fp8. We set `use_deep_seek_fp8=False` for all other cases like per-tensor FP8, nvfp4, mxfp4,etc.

File: csrc/trtllm_fused_moe_kernel_launcher.cu (hunk 4 above)
Comment (ChristinaZ) [reply csrc/trtllm_fused_moe_kernel_launcher.cu]: Yes, agree. It's better that we can add comments on them.

File: csrc/trtllm_fused_moe_kernel_launcher.cu (hunk 3 above)
Comment (pavanimajety) [reply csrc/trtllm_fused_moe_kernel_launcher.cu:194]: Does it make more sense to make it more explicit by making it an Enum? Probably for future, not in this PR

**Comment by wenscarl** [comment]
@yzh119 could you take a look and merge it? Thanks!
